### PR TITLE
Add CumulativeToDelta Processor

### PR DESCRIPTION
**Pull request by EddieEldridge**
* Addresses: https://github.com/SumoLogic/sumologic-otel-collector/issues/595

* [Cumulative To Delta Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/cumulativetodeltaprocessor)

* My use case of this is that not all vendors support Cumulative metrics and being able to use this processor to convert them to Delta would be incredibly helpful.

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
@EddieEldridge thank you for the contribution :)

**Comment by EddieEldridge** [comment]
> It should be rather added here (bottom part of snippet):
> 
> https://github.com/SumoLogic/sumologic-otel-collector/blob/956e10308aedeec3622bf4ff30afc6866d1ad4df/otelcolbuilder/.otelcol-builder.yaml#L34-L69
> 
> as `path: ./../pkg/processor/cumulativetodeltaprocessor` refers to this repository code

Woops, my bad. Should be fixed now :)

**Comment by sumo-drosiek** [comment]
and [readme](https://github.com/SumoLogic/sumologic-otel-collector#components)

**Comment by sumo-drosiek** [comment]
@EddieEldridge Crucial part of readme is the table with list of supported components. Unfortunately this is alphabetical as well :sweat:

**Comment by EddieEldridge** [comment]
> @EddieEldridge Crucial part of readme is the table with list of supported components. Unfortunately this is alphabetical as well 😓
> 
> I'm personally using this plugin to format table: https://marketplace.visualstudio.com/items?itemName=darkriszty.markdown-table-prettify

Ok, after much struggle I think I've managed to add that in without messing up the whole table. Thanks for being patient :D

File: README.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ The rest of the components in the table are pure upstream OpenTelemetry componen
 [zookeeperreceiver]: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.51.0/receiver/zookeeperreceiver
 
 [attributesprocessor]: https://github.com/SumoLogic/opentelemetry-collector-contrib/tree/v0.51.0-filterprocessor/processor/attributesprocessor
+[cumulativetodeltaprocessor]: https://github.com/SumoLogic/opentelemetry-collector-contrib/tree/v0.51.0-filterprocessor/processor/cumulativetodeltaprocessor
```
Comment (swiatekm-sumo) [inline README.md:108]: Shouldn't this url point to upstream?

**Comment by swiatekm-sumo** [comment]
@EddieEldridge I think we're good to go after you rebase and resolve conflicts. It'd be best if you squashed it all to a single commit and followed https://www.conventionalcommits.org/en/v1.0.0/ for the message.

**Comment by EddieEldridge** [comment]
> 

Cool, I think that should be good now. Let me know if there's anything else that needs changing :)

**Review by swiatekm-sumo — APPROVED** [review]
👍

**Pull request merged**